security/advancedtls/advancedtls_integration_test.go

```diff
@@ -408,3 +411,272 @@ func (s) TestEnd2End(t *testing.T) {
+				if err != nil {
+					t.Fatalf("copyFileContents(%s, %s) failed: %v", testdata.Path("server_trust_cert_2.pem"), serverTrustTmp.Name(), err)
+				}
+			},
+		},
+		{
+			desc: "TestServerPeerCertProviderClientTrustCertProvider",
+			certUpdateFunc: func() {
+				err = copyFileContents(testdata.Path("server_cert_2.pem"), serverCertTmp.Name())
+				if err != nil {
+					t.Fatalf("copyFileContents(%s, %s) failed: %v", testdata.Path("server_cert_2.pem"), serverCertTmp.Name(), err)
+				}
+			},
+			keyUpdateFunc: func() {
+				err = copyFileContents(testdata.Path("server_key_2.pem"), serverKeyTmp.Name())
+				if err != nil {
+					t.Fatalf("copyFileContents(%s, %s) failed: %v", testdata.Path("server_key_2.pem"), serverKeyTmp.Name(), err)
+				}
+			},
+			trustCertUpdateFunc: func() {
+				err = copyFileContents(testdata.Path("client_trust_cert_2.pem"), clientTrustTmp.Name())
+				if err != nil {
+					t.Fatalf("copyFileContents(%s, %s) failed: %v", testdata.Path("client_trust_cert_2.pem"), clientTrustTmp.Name(), err)
+				}
+			},
+		},
+	} {
+		test := test
+		t.Run(test.desc, func(t *testing.T) {
+			// Copy the credential contents to the temporary files.
+			err = copyFileContents(testdata.Path("client_cert_1.pem"), clientCertTmp.Name())
+			if err != nil {
+				t.Fatalf("copyFileContents(%s, %s) failed: %v", testdata.Path("client_cert_1.pem"), clientCertTmp.Name(), err)
+			}
+			err = copyFileContents(testdata.Path("client_key_1.pem"), clientKeyTmp.Name())
+			if err != nil {
+				t.Fatalf("copyFileContents(%s, %s) failed: %v", testdata.Path("client_key_1.pem"), clientKeyTmp.Name(), err)
+			}
+			err = copyFileContents(testdata.Path("client_trust_cert_1.pem"), clientTrustTmp.Name())
+			if err != nil {
+				t.Fatalf("copyFileContents(%s, %s) failed: %v", testdata.Path("client_trust_cert_1.pem"), clientTrustTmp.Name(), err)
+			}
+			err = copyFileContents(testdata.Path("server_cert_1.pem"), serverCertTmp.Name())
+			if err != nil {
+				t.Fatalf("copyFileContents(%s, %s) failed: %v", testdata.Path("server_cert_1.pem"), serverCertTmp.Name(), err)
+			}
+			err = copyFileContents(testdata.Path("server_key_1.pem"), serverKeyTmp.Name())
+			if err != nil {
+				t.Fatalf("copyFileContents(%s, %s) failed: %v", testdata.Path("server_key_1.pem"), serverKeyTmp.Name(), err)
+			}
+			err = copyFileContents(testdata.Path("server_trust_cert_1.pem"), serverTrustTmp.Name())
+			if err != nil {
+				t.Fatalf("copyFileContents(%s, %s) failed: %v", testdata.Path("server_trust_cert_1.pem"), serverTrustTmp.Name(), err)
+			}
+			// Create PEMFileProvider(s) watching the content changes of temporary
+			// files.
+			clientIdentityProvider, err := NewPEMFileProvider(PEMFileProviderOptions{
+				CertFile:         clientCertTmp.Name(),
+				KeyFile:          clientKeyTmp.Name(),
+				IdentityInterval: 200 * time.Millisecond,
+			})
+			if err != nil {
+				t.Errorf("NewPEMFileProvider failed: %v", err)
```

Comment:
Please output the options passed to the `NewPEMFileProvider` in the error message.